pkg/commands/workload.go

```diff
@@ -771,29 +777,69 @@ func isUrl(str string) (bool, error) {
 	}
 }
 
-func (opts *WorkloadOptions) GetReadyConditionWorkers(c *cli.Config, workload *cartov1alpha1.Workload, workers []wait.Worker) []wait.Worker {
-	workers = append(workers, func(ctx context.Context) error {
+func raceWithTimeout(ctx context.Context, c *cli.Config, workload *cartov1alpha1.Workload, timeout time.Duration, shouldPrint bool, errMsg string, workers []wait.Worker) error {
+	err := wait.Race(ctx, timeout, workers)
+	// print wait error only if output is not set or it was not used with --yes
+	if err != nil {
+		if err == context.DeadlineExceeded {
+			cli.PrintPrompt(shouldPrint, c.Printf, "%s timeout after %s waiting for %q to become ready\n", printer.Serrorf(fmt.Sprintf("%s:", errMsg)), timeout, workload.Name)
+		} else {
+			cli.PrintPrompt(shouldPrint, c.Eprintf, "%s %s\n", printer.Serrorf(fmt.Sprintf("%s:", errMsg)), err)
+		}
+	}
+	return err
+}
+
+func getStatusChangeWorkers(c *cli.Config, workload *cartov1alpha1.Workload) wait.Worker {
+	worker := wait.Worker(func(ctx context.Context) error {
+		previousReadyCond := printer.FindCondition(workload.Status.Conditions, cartov1alpha1.WorkloadConditionReady)
+		clientWithWatch, err := watch.GetWatcher(ctx, c)
+		if err != nil {
+			return err
+		}
+		return wait.UntilCondition(ctx, clientWithWatch, types.NamespacedName{Name: workload.Name, Namespace: workload.Namespace}, &cartov1alpha1.WorkloadList{}, func(target client.Object) (bool, error) {
+			obj, ok := target.(*cartov1alpha1.Workload)
+			if !ok {
+				return false, nil
+			}
+			if obj.Generation != obj.Status.ObservedGeneration {
+				return false, nil
+			}
+			currentReadyCond := printer.FindCondition(obj.Status.Conditions, cartov1alpha1.WorkloadConditionReady)
+			if previousReadyCond != nil && currentReadyCond != nil {
+				if previousReadyCond.LastTransitionTime.Before(&currentReadyCond.LastTransitionTime) {
+					return true, nil
+				}
+			}
+			return false, nil
+		})
+	})
+	return worker
+}
+
+func getReadyConditionWorkers(c *cli.Config, workload *cartov1alpha1.Workload) wait.Worker {
```

Comment:
*nit:* How about calling this `getReadyConditionWorker` as return type is `wait.Worker` and not an []?